### PR TITLE
Assignment 4

### PR DESCRIPTION
**Pull request by glopez42**
Assignment 4 done including task 8 & 9. 
Guillermo López García

### CODE REVIEW

File: Assignment4/glopez42-180182/task06.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.4: Add to the individual JaneSmith the email address, fullName, given and family names**"""
+
+print("**TASK 6.4**")
+g.add((ns.JaneSmith, VCARD.EMAIL, Literal("jane.smith@mail.com")))
+g.add((ns.JaneSmith, VCARD.FN, Literal("Jane Smith")))
+g.add((ns.JaneSmith, VCARD.GIVEN, Literal("Jane")))
```
Comment (dgarijo) [inline Assignment4/glopez42-180182/task06.py:54]: The right property is `Given`, not `GIVEN`. I may  have told this wrong a couple of times, it's always good to check with the vocab.

File: Assignment4/glopez42-180182/task07.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+print("**TASK 7.1**")
+
+print("Using rdflib:")
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
+  print(s)
+
+print("Using SPARQL:")
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE { 
+    ?Subject rdfs:subClassOf ns:LivingThing.
```
Comment (dgarijo) [inline Assignment4/glopez42-180182/task07.py:32]: You need to consider the subclasses of subclasses. Use `*` or `+`

File: Assignment4/glopez42-180182/task07.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+print()
+
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+print("**TASK 7.2**")
+
+print("Using rdflib:")
+subclasses = [s for s,p,o in g.triples((None, RDFS.subClassOf, ns.Person))]
```
Comment (dgarijo) [inline Assignment4/glopez42-180182/task07.py:50]: same issue as above. Hint: make a recursive function

File: Assignment4/glopez42-180182/task07.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+      print(s)
+
+print("Using SPARQL:")
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE { 
+    {
+        ?Subject rdf:type ns:Person. 
+    }
+    UNION {
+    ?class rdfs:subClassOf ns:Person.
```
Comment (dgarijo) [inline Assignment4/glopez42-180182/task07.py:64]: same problem as above

File: Assignment4/glopez42-180182/task07.py [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+"""
+print("**TASK 7.2**")
+
+print("Using rdflib:")
+subclasses = [s for s,p,o in g.triples((None, RDFS.subClassOf, ns.Person))]
+subclasses.append(ns.Person)
+
+for object in subclasses:
+    for s,p,o in g.triples((None, RDF.type, object)):
+      print(s)
```
Comment (dgarijo) [inline Assignment4/glopez42-180182/task07.py:55]: same problem as above

File: Assignment4/glopez42-180182/task07.py [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE { 
+    {
+        ?Subject foaf:knows ?o1.
+        ?Subject foaf:knows ?o2.
+        FILTER (?o1 != ?o2)
+    }
+  }
+  GROUP BY ?Subject
```
Comment (dgarijo) [inline Assignment4/glopez42-180182/task07.py:176]: Group By not needed!

**Review by dgarijo — CHANGES_REQUESTED** [review]
Please address the remaining small changes

**Comment by glopez42** [comment]
Thanks, I have adressed those changes and uploaded the new version.
 Greetings

**Review by dgarijo — APPROVED** [review]

**Comment by dgarijo** [comment]
Perfect

**Pull request merged**